### ISSUE TITLE
music script for dwmblocks doesn't run until killing dwmblocks

### ISSUE THREAD

**Issue by paniash** [LukeSmithxyz/voidrice#747]
As the title suggests, when I boot up my laptop, the music script for dwmblocks doesn't run until I kill and restart dwmblocks. Maybe it has to do with my `.xinitrc` or `.xprofile` but I'm not sure. I've tried tinkering with the order in which applications start but to no avail. Here are the files:

`.xinitrc`
```
#!/bin/sh

export STATUSBAR="dwmblocks"

. ~/.xprofile

# Here we start dwm
# The loop is just to enable dwm's "restart" feature

while :; do
	ssh-agent dwm || break
done	
```

`.xprofile`
```
#!/bin/sh

# This file runs when a DM logs into a graphical session
# If you use startx/xinit like a Chad, this file will also be source

dwmblocks &	# statusbar
mpd &		# starts music player daemon
picom &		# starts compositor
dunst &		# starts notification daemon
remaps &	# maps caplock to escape
xwallpaper --zoom /home/ashish/.config/wall.png & 	# sets wallpaper
nm-applet & 	# starts nm-applet
unclutter &	# removes mouse cursor when not in use
```

Thanks in advance!

**Comment by AlexSyssoyev** [comment]
Once you have added music to the playlist in nmcpcpp do you use mod+p to start the music?

For me to get the script to run i had to add a ```;pkill -RTMIN+11 dwmblocks``` to the command that runs when using mod+p. So it ended up looking like this:
```
	{ MODKEY,			XK_p,			spawn,		SHCMD("mpc toggle; pkill -RTMIN+11 dwmblocks") },
```

**Comment by paniash** [comment]
```
	{ 0,                       XF86XK_AudioPlay,             spawn,          SHCMD("mpc toggle; pkill -RTMIN+11 dwmblocks") },
```
I use this and once I restart dwmblocks, it works fine (playing or pausing in ncmpcpp updates my music module). The issue occurs only on bootup. It works fine after I kill and restart dwmblocks.

**Comment by AlexSyssoyev** [comment]
Maybe adding a sleep 5 before dwmblocks in xprofile? It will fix it but it would be just a sloppy workaround
``` sleep 5 && dwmblocks```

**Comment by paniash** [comment]
Thanks @AlexSyssoyev for the suggestion. The hack that you suggested does work! However, I seem to have run into a minor issue. The 'mpdup' script doesn't run until I kill and restart dwmblocks. Mind the music script works fine now but when I change music through ncmpcpp, the music module doesn't update until I kill and restart dwmblocks (only when I boot up), 'mpdup' works fine after killing and restarting dwmblocks though.

**Comment by narukeh** [comment]
@paniash  have a read of #663 to give you any idea.
if you comment out line number 5 of `music` script : ```pidof -x mpdup >/dev/null 2>&1 || mpdup >/dev/null 2>&1 &``` i bet you will not have a problem, but then it will not update when a song is over and the next one starts. tho i works when you change the song manually. 
do you have a old larbs install? `[ larbs -eq old ] && "reinstall larbs"`  😛😛 make sure you have a fresh larbs.

**Comment by paniash** [comment]
This is my music script:
```
#!/bin/sh

# Displays the current song being played on mpd

pidof -x mpdup >/dev/null 2>&1 || mpdup >/dev/null 2>&1 &

ARTIST=$(mpc current -f %artist%)
SONG=$(mpc current -f %title%)
STATUS=$(mpc status | sed -n 2p | awk '{print $1}')

if [ ! $SONG ]; then
	echo ""
else
    if [ "$STATUS" = "[playing]" ]; then
			ICON=""
    else
			ICON=""
    fi
			
	echo "$ICON $ARTIST - $SONG "
fi
```

Commenting out the `mpdup` line won't even use the `mpdup` script so, no more automatic update of the music module on changing songs in ncmpcpp.

No, I don't use larbs, though I use a few of Luke's scripts.

**Comment by narukeh** [comment]
@paniash  
TLDR: use luke's script, or the one i wrote.

first: does the same happen when u use luke's [music](https://github.com/LukeSmithxyz/voidrice/blob/master/.local/bin/statusbar/music) script ?
second: please don't write such ugly code ever again, use this: 
```
#!/bin/sh

pidof -x mpdup >/dev/null 2>&1 || mpdup >/dev/null 2>&1 &

ARTIST=$(mpc current -f %artist%)
SONG=$(mpc current -f %title%)
STATUS=$(mpc status | sed -n 2p | awk '{print $1}')

emoji() {
	[ "$STATUS" = "[playing]" ] && ICON="😁" || ICON="🥵"
}

[ -n "$SONG" ] && { emoji && echo "$ICON $ARTIST - $SONG " ;}
```
its your script, but pretty, also your emoji didn't render, so i used two random, change them.

**Comment by paniash** [comment]
I'm still learning shell scripting so thank you @narukeh for correcting me with your "constructive" second comment. I do appreciate the changes that you suggested however you could be a little kinder with your words.

Also the icons didn't render because they are part of font-awesome :P

**Comment by narukeh** [comment]
@paniash  (i guess a joking sarcastic intonation doesn't get transmitted through text, at least not without emojis)
so is the problem solved? you didnt answer my first question.

**Comment by paniash** [comment]
Yes, it seems that the issue was with the mpdup script (atleast for me).
I changed this line as follows:
```
mpc idle >/dev/null && pkill -RTMIN+11 dwmblocks || break
```

And now it works flawlessly. Thanks!

**Comment by narukeh** [comment]
@paniash  great! dont forget to close the issue.
